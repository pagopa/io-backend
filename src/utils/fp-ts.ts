import { and as simpleAnd, Predicate } from "fp-ts/lib/function";

export const and = <A>(
  p1: Predicate<A>,
  ...predicates: ReadonlyArray<Predicate<A>>
) => {
  return predicates.reduce((acc, predicate) => {
    return simpleAnd(acc, predicate);
  }, p1);
};
