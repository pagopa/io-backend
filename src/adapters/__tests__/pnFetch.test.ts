import * as E from "fp-ts/Either";



describe("pathParamsFromUrl", () => {

});