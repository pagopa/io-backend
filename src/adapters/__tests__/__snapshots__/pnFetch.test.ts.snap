// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pathParamsFromUrl decode a not vaild simple test 1`] = `
Array [
  Object {
    "context": Array [
      Object {
        "actual": "/tests/12345",
        "key": "",
        "type": Type {
          "decode": [Function],
          "encode": [Function],
          "is": [Function],
          "name": "pathParamsFromUrl",
          "validate": [Function],
        },
      },
    ],
    "message": "input is not a valid /^\\\\/test\\\\/([^/]+)$/",
    "value": "/tests/12345",
  },
]
`;
